src/libecalc/common/priority_optimizer.py

```diff
@@ -80,15 +87,15 @@ def optimize(
 
                 # Register the valid timesteps as valid and keep track of the operational setting used
                 is_valid[new_valid_indices] = True
-                priorities_used[new_valid_indices] = priority_index
+                priorities_used[new_valid_indices] = priority_name
 
                 if all(is_valid.values):
                     # quit as soon as all time-steps are valid. This means that we do not need to test all settings.
                     break
-                elif priority_index + 1 == len(priorities):
+                elif list(priorities.keys()).index(priority_name) + 1 == len(priorities):
```

Comment:
funksjon for å hente riktig id? mtp index+1?

Reply:
Kunne vel droppet denne hvis vi initialiserte til siste pri?